### PR TITLE
Simplify SSR tag removing, also fix #44

### LINKED ISSUE #44: Error when meta content includes quote
Version: v3.0.0
Error stack: 
```
index.esm.js?d732:58 Uncaught DOMException: Failed to execute 'querySelector' on 'Element': 'meta[property="og:title"][content=""Có tâm" như Lý Quý Khánh chụp ảnh cho Hà Hồ: Lần nào cũng hết hồn chim én!"][data-rh=""]' is not a valid selector.
    at HeadTag.componentDidMount (webpack-internal:///./node_modules/react-head/dist/index.esm.js:74:33)
    at commitLifeCycles (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:14361:22)
    at commitAllLifeCycles (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:15462:7)
    at HTMLUnknownElement.callCallback (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:100:14)
    at Object.invokeGuardedCallbackDev (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:138:16)
    at invokeGuardedCallback (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:187:29)
    at commitRoot (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:15603:7)
    at completeRoot (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:16618:34)
    at performWorkOnRoot (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:16563:9)
    at performWork (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:16482:7)
```

Quite serious that page just gone white.

It seems cause of meta content which include double quote mark(")
Maybe do not use content as part of querySelector?  (or perform suitable escaping before query)

And, I only want head tags performs on server, and leave it alone at client side (My MPA do not use front-ending routing), Maybe add some option to do this.

### PR DESCRIPTION
**Pull request by zhangciwu**

### CODE REVIEW

**Comment by TrySound** [comment]
I don't think this operation should happen in every `HeadTag`. We can do it once in `HeadProvider` and make it required.

File: src/HeadTag.js [hunk 1]
```diff
@@ -19,14 +18,11 @@ export default class HeadTag extends React.Component {
...
     this.setState({ canUseDOM: true });
 
-    const { tag, children, ...rest } = this.props;
-    const ssrTags = document.head.querySelector(
-      `${tag}${buildSelector(rest)}[data-rh=""]`
-    );
+    const { tag, ...rest } = this.props;
+    const ssrTags = document.head.querySelectorAll(`[data-rh=""]`);
+
+    ssrTags.forEach(e => e.remove());
```
[the remark this answers is not in the document]
Comment (zhangciwu) [reply src/HeadTag.js]: querySelectorAll will get blank list if no match, so need not to do null check

File: src/__tests__/HeadTags.web.test.js [hunk 2]
```diff
@@ -30,14 +43,10 @@ describe('HeadTag during client', () => {
...
-      'test[name="x"][content="testing"][data-rh=""]'
-    );
-    expect(qsMock).toHaveBeenCalledWith('title[data-rh=""]');
-    expect(qsMock).toHaveBeenCalledWith('style[data-rh=""]');
-    expect(qsMock).toHaveBeenCalledWith('link[href="index.css"][data-rh=""]');
-    expect(qsMock).toHaveBeenCalledWith('meta[charset="utf-8"][data-rh=""]');
+    expect(qsMock).toHaveBeenCalledWith('[data-rh=""]');
+    expect(qsMock).toHaveBeenCalledTimes(5);
     expect(removeMock).toHaveBeenCalledTimes(5);
+    // first document.head.querySelectorAll removes all ssr nodes, rest just got blank Nodelist
```
[the remark this answers is not in the document]
Comment (zhangciwu) [reply src/__tests__/HeadTags.web.test.js]: So does true dom, querySelectorAll get all of ssr heads,

**Comment by zhangciwu** [comment]
@TrySound Yeah, let me see it.

**Comment by zhangciwu** [comment]
Got a problem, if make `HeadProvider` required at front, there will be two scene:
- `Route` inside `HeadProvider` : `HeadProvider ` will be unaware of page switching, making it's responsibility of checking duplications of tags(title, etc.) fail to work.
- `HeadProvider` inside `Route` : actually not possible, cause `Route` is inside isomorphic code(like `example/src/App.js`), making `HeadProvider` inside it will cause server side `HeadProvider` fail.

So I think just don't use `HeadProvider` at front, removing SSR tags can be done just once by adding some `removedFlag` and check if before real actions. Or just put the removing procedure outside `HeadTag` class declaration, Making it invoking at import time (at front).

How do you think @TrySound

**Comment by TrySound** [comment]
Closed in favour https://github.com/tizmagik/react-head/pull/55. Thanks for your work.

**Pull request closed without merging**